src-plugins/itkFilters/itkFiltersBinaryThresholdingProcess.cpp

```diff
@@ -0,0 +1,142 @@
+     : itkFiltersProcessBase(other), d(other.d)
+{
+}
+
+//-------------------------------------------------------------------------------------------
+
+itkFiltersBinaryThresholdingProcess::~itkFiltersBinaryThresholdingProcess( void )
+{
+    delete d;
+}
+
+//-------------------------------------------------------------------------------------------
+
+bool itkFiltersBinaryThresholdingProcess::registered( void )
+{
+    return dtkAbstractProcessFactory::instance()->registerProcessType("itkBinaryThresholdingProcess", createitkFiltersBinaryThresholdingProcess);
+}
+
+//-------------------------------------------------------------------------------------------
+
+QString itkFiltersBinaryThresholdingProcess::description() const
+{
+    return tr("ITK binary thresholding filter");
+}
+
+//-------------------------------------------------------------------------------------------
+
+void itkFiltersBinaryThresholdingProcess::setParameter(int data, int channel)
+{
+    if ( channel > 1 )
+    {
+        return;
+    }
+    if (channel == 0)
+        d->insideValue = data;
+    if (channel == 1)
+        d->outsideValue = data;
+}
+
+void itkFiltersBinaryThresholdingProcess::setParameter(double data, int channel)
+{
+    if ( channel > 1 )
+    {
+        return;
+    }
+    if (channel == 0)
+        d->lowerValue = data;
+    if (channel == 1)
+        d->upperValue = data;
+}
+
+//-------------------------------------------------------------------------------------------
+
+int itkFiltersBinaryThresholdingProcess::tryUpdate()
+{   
+    int res = DTK_FAILURE;
+
+    if ( getInputData() )
+    {
+        res = DISPATCH_ON_3D_PIXEL_TYPE(&itkFiltersBinaryThresholdingProcess::updateProcess, this, getInputData());
+    }
+
+    return res;
+}
+
+template <class ImageType>
+int itkFiltersBinaryThresholdingProcess::updateProcess(medAbstractData* inputData)
+{
+    typename ImageType::Pointer inputImage = static_cast<ImageType*>(inputData->data());
+
+    typedef itk::BinaryThresholdImageFilter < ImageType, ImageType>  BinaryThresholdImageFilterType;
+    typename BinaryThresholdImageFilterType::Pointer thresholdFilter = BinaryThresholdImageFilterType::New();
+    thresholdFilter->SetInput(inputImage);
+    thresholdFilter->SetLowerThreshold(d->lowerValue);
+    thresholdFilter->SetUpperThreshold(d->upperValue);
+    thresholdFilter->SetInsideValue(d->insideValue);
+    thresholdFilter->SetOutsideValue(d->outsideValue);
+
+    itk::CStyleCommand::Pointer callback = itk::CStyleCommand::New();
+    callback->SetClientData ( ( void * ) this );
+    callback->SetCallback ( itkFiltersProcessBase::eventCallback );
+    thresholdFilter->AddObserver ( itk::ProgressEvent(), callback );
+
+    thresholdFilter->Update();
+
+    getOutputData()->setData ( thresholdFilter->GetOutput() );
+
+    QString newSeriesDescription = "binaryThreshold " + QString::number(d->insideValue);
```

Comment:
since there are 4 parameters, we can perhaps remove this one from the description? Just "binaryThreshold"